client/state/themes/selectors.js

```diff
@@ -603,3 +603,26 @@ export function isThemePurchased( state, themeId, siteId ) {
 	const sitePurchases = getSitePurchases( state, siteId );
 	return some( sitePurchases, { productSlug: 'premium_theme', meta: themeId } );
 }
+
+/**
+ * Returns themeId to preview for site.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {Object}         Last query
+ */
+export function getThemeForPreviewData( state ) {
+	const themeData = get( state.themes.previewingTheme, 'themeData', null );
+	const themeOptions = get( state.themes.previewingThemeOptions, 'themeOptions', null );
+	return { themeData, themeOptions };
+}
+
+/**
+ * Returns the ThemePreview state
+ *
+ * @param  {Object}  state Global state tree
+ * @return {?String}  ThemePreview state
+ */
+export function getThemePreviewState( state ) {
```

Comment:
Can we turn this into a bool ? For one, I'd rather avoid `State` in the name of selectors and reducers, since, well, they're all about state 😉 And OTOH, I think this thing can only ever assume one of two possible values, right? You could name the selector `isThemePreviewVisible`, and the reducer `themePreviewVisibility`.